backend/django/core/views/api_annotate.py

```diff
@@ -121,6 +121,28 @@ def label_distribution_inverted(request, project_pk):
     return Response(dataset)
 
 
+@api_view(["POST"])
+@permission_classes((IsCoder,))
+def unassign_data(request, data_pk):
+    """Take a datum that is in the assigneddata queue for that user and remove it from
+    the assignedData queue.
+
+    Args:
+        request: The POST request
+        data_pk: Primary key of the data
+    Returns:
+        {}
+    """
+    data = Data.objects.get(pk=data_pk)
+    profile = request.user.profile
+    response = {}
+
+    assignment = AssignedData.objects.get(data=data, profile=profile)
+    assignment.delete()
+
```

Comment:
Need to add the check_and_trigger_model call which is also at the bottom of skip_data. This will refill their deck if they've skipped their last card.

Reply:
Good catch!

Reply:
Actually twas a batch catch all along. This function checks if the queue is empty, but of course you're not messing with the queue at all. You actually don't need this. A simple refresh of the page will give them new cards.